java/code/src/com/redhat/rhn/domain/contentmgmt/PackageFilter.java

```diff
@@ -87,10 +87,10 @@ private static <T> T getField(Package pack, String field, Class<T> type) {
 
     private static boolean checkNameAndArch(String field, String value, Package pack) {
         if (field.equals("nevr")) {
-            return value.replaceAll("(.*)-(.*:)?(.*)-(.*)", "$1").equals(pack.getPackageName().getName());
+            return value.replaceAll("^(.*)-([^:]*+:)?([^-]*+)-([^-]*+)$", "$1").equals(pack.getPackageName().getName());
```

Comment:
Why don't you use `[^-]*` for the first group? I suppose that one could backtrack too, right? The last group can probably still be `.*`

Reply:
that makes sense